app/src/main/java/com/humberto/tasky/core/database/data_source/RoomLocalAgendaDataSource.kt

```diff
@@ -0,0 +1,131 @@
+package com.humberto.tasky.core.database.data_source
+
+import android.database.sqlite.SQLiteFullException
+import com.humberto.tasky.core.database.dao.AgendaDao
+import com.humberto.tasky.core.database.dao.EventDao
+import com.humberto.tasky.core.database.dao.ReminderDao
+import com.humberto.tasky.core.database.dao.TaskDao
+import com.humberto.tasky.core.database.mapper.toEvent
+import com.humberto.tasky.core.database.mapper.toEventEntity
+import com.humberto.tasky.core.database.mapper.toReminder
+import com.humberto.tasky.core.database.mapper.toReminderEntity
+import com.humberto.tasky.core.database.mapper.toTask
+import com.humberto.tasky.core.database.mapper.toTaskEntity
+import com.humberto.tasky.core.domain.agenda.Agenda
+import com.humberto.tasky.core.domain.agenda.LocalAgendaDataSource
+import com.humberto.tasky.core.domain.event.Event
+import com.humberto.tasky.core.domain.reminder.Reminder
+import com.humberto.tasky.core.domain.task.Task
+import com.humberto.tasky.core.domain.util.DataError
+import com.humberto.tasky.core.domain.util.Result
+import com.humberto.tasky.core.domain.util.toEndOfDayUtc
+import com.humberto.tasky.core.domain.util.toStartOfDayUtc
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.map
+import java.time.LocalDate
+
+class RoomLocalAgendaDataSource(
+    private val agendaDao: AgendaDao,
+    private val taskDao: TaskDao,
+    private val eventDao: EventDao,
+    private val reminderDao: ReminderDao,
+): LocalAgendaDataSource {
+    private suspend fun upsertTasks(
+        tasks: List<Task>
+    ): Result<Unit, DataError.Local> {
```

Comment:
For return types with an empty result, I like the following alias: `typealias EmptyResult<E> = Result<Unit, E>`